server/src/main/java/org/jetbrains/bsp/bazel/server/BloopExporter.java

```diff
@@ -0,0 +1,549 @@
+      var digest = MessageDigest.getInstance("MD5");
+      digest.update(label.toString().getBytes(StandardCharsets.UTF_8));
+      return "z_" + BaseEncoding.base16().encode(digest.digest()).substring(0, 12);
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private Option<Config.Scala> extractScalaModuleFromLanguageData(LanguageData data) {
+    if (!(data instanceof ScalaModule)) {
+      return Option.none();
+    }
+
+    var scalaMod = (ScalaModule) data;
+    var config =
+        new Config.Scala(
+            scalaMod.sdk().organization(),
+            "scala-compiler",
+            scalaMod.sdk().version(),
+            CollectionConverters.asScala(scalaMod.scalacOpts()).toList(),
+            CollectionConverters.asScala(scalaMod.sdk().compilerJars().map(Paths::get)).toList(),
+            scala.Option.empty(),
+            scala.Some.apply(
+                Config.CompileSetup$.MODULE$.apply(
+                    Config.Mixed$.MODULE$, true, false, false, true, true)));
+    return Option.some(config);
+  }
+
+  private Seq<String> sanitizeJavacOpts(Seq<String> opts) {
+    return opts.filter(p -> !BAD_JAVAC_OPTS.contains(p));
+  }
+
+  private Option<Config.Java> extractJavaModuleFromLanguageData(LanguageData languageData) {
+    var javaModule = javaModuleFromLanguageData(languageData);
+    return javaModule.map(
+        mod ->
+            new Config.Java(
+                CollectionConverters.asScala(sanitizeJavacOpts(mod.javacOpts())).toList()));
+  }
+
+  private <T> scala.Option<T> toScalaOption(Option<T> opt) {
+    return opt.fold(scala.Option::empty, Some::apply);
+  }
+
+  private Seq<Path> fixupClassPath(Seq<URI> input) {
+    return input
+        .iterator()
+        .map(p -> this.localArtifacts.getOrElse(p, p))
+        .distinct()
+        .map(Paths::get)
+        .toArray();
+  }
+
+  private Option<Config.Platform> extractPlatformFromLanguageData(LanguageData languageData) {
+    var javaModule = javaModuleFromLanguageData(languageData);
+    return javaModule.map(
+        mod -> {
+          var runtimeJdk = mod.runtimeJdk().getOrElse(mod.jdk());
+          var jvmConfig =
+              new Config.JvmConfig(
+                  toScalaOption(runtimeJdk.javaHome().map(Paths::get)),
+                  CollectionConverters.asScala(mod.jvmOps()).toList());
+          var classPath =
+              CollectionConverters.asScala(fixupClassPath(mod.runtimeClasspath())).toList();
+
+          return new bloop.config.Config$Platform$Jvm(
+              jvmConfig,
+              toScalaOption(mod.mainClass()),
+              scala.Some.apply(jvmConfig),
+              scala.Some.apply(classPath),
+              scala.Option.empty());
+        });
+  }
+
+  private Config.Test bloopTestFramework() {
+    var framework =
+        new Config.TestFramework(
+            CollectionConverters.asScala(
+                    Collections.singleton("munit.internal.junitinterface.PantsFramework"))
```

Comment:
Are you sure this test interface will be on the classpath? It is normally provided by `org.scalameta:junit-interface:1.0.0-M3`. Some test classes will not be detected as such if the test interface is not on the (runtime) classpath.